geoportal/c2cgeoportal_geoportal/views/i18n.py

```diff
@@ -0,0 +1,22 @@
+from pyramid.httpexceptions import HTTPFound
+from pyramid.view import view_config
+
+from c2cgeoportal_geoportal.lib.cacheversion import get_cache_version
+
+
+@view_config(route_name='localejson')
+def locale(request):
+    return HTTPFound(
+        request.static_url(
+            '/etc/geomapfish/static/{lang}.json'.format(
+                package=request.registry.settings["package"],
+                lang=request.locale_name,
+                _query={
+                    'cache': get_cache_version(),
+                }
+            )
+        ),
+        headers={
+            "Access-Control-Allow-Origin": "*",
```

Comment:
Do we relay needs this header?

Reply:
For the API it seems we need it, as it will be requested from CMS urls.

I cannot use `c2cgeoportal_geoportal.caching.set_common_headers` directly with which takes a response object as parameter with an HTTPFound exception.

Ideally CORS headers should threated in a tween or view decorator, not with an imperative method as it is done with `set_common_headers`.

We can also create a response object with info.registry.queryAdapterOrSelf(result, IResponse) as it is done by pyramid : https://github.com/Pylons/pyramid/blob/3af1883bcd617d74eb8c9b134b5ac830f8cdd2a9/src/pyramid/viewderivers.py#L408

We may also have a look at https://cornice.readthedocs.io/en/latest/index.html

Reply:
I agree with all of that but for the final file, do we also need that for the HTTPFound responce? Do you test it without this header?
When I don some search about that I found: https://stackoverflow.com/questions/18539403/chrome-cancels-cors-xhr-upon-http-302-redirect, and it seems that we can't have a redirect on an OPTION request...
For me we should do a test to be sure that it's working correctly ...

Reply:
I alreaddy tested it from ngeo make serve-api from another port than demo_geomapfish, and it does not works without CORS header.

Reply:
OK thanks :-)